collection.go

```diff
@@ -136,7 +136,7 @@ func NewCollectionWithOptions(spec *CollectionSpec, opts CollectionOptions) (col
 		var handle *btf.Handle
 		if mapSpec.BTF != nil {
 			handle, err = loadBTF(btf.MapSpec(mapSpec.BTF))
-			if err != nil && !btf.IsNotSupported(err) {
+			if err != nil && !xerrors.Is(err, btf.ErrNotSupported) {
```

Comment:
as `ErrNotSupported` is defined in `prog.go`, we can use it directly.

Reply:
True, but I wanted to keep them separate if we need to change the semantics later. My idea was that btf.ErrNotSupported at some point may be != ErrNotSupported. Maybe it's overkill, not sure.

Reply:
sounds good to me.